pkg/apis/configuration/v1/types.go

```diff
@@ -313,3 +314,43 @@ type VirtualServerRouteStatus struct {
 	ReferencedBy      string             `json:"referencedBy"`
 	ExternalEndpoints []ExternalEndpoint `json:"externalEndpoints,omitempty"`
 }
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:validation:Optional
+// +kubebuilder:resource:shortName=pol
+
+// Policy defines a Policy for VirtualServer and VirtualServerRoute resources.
+type Policy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec PolicySpec `json:"spec"`
+}
+
+// PolicySpec is the spec of the Policy resource.
+// The spec includes multiple fields, where each field represents a different policy.
+// Only one policy (field) is allowed.
+type PolicySpec struct {
+	AccessControl *AccessControl        `json:"accessControl"`
+	RateLimit     *v1alpha1.RateLimit   `json:"rateLimit"`
```

Comment:
shall we move those types from `v1alpha1` to this package as well?  since they belong to this type and can't exist without it. we can add a comment that those fields are an alpha feature

Reply:
Done in https://github.com/nginxinc/kubernetes-ingress/pull/1290/commits/0f347250cb8719e7de409e5f510c4d1470e386c2